docs/user-guide/config-best-practices.md

```diff
@@ -0,0 +1,47 @@
+1. Create a service before corresponding replication controllers so that the scheduler can spread the pods comprising the service. #11144 (or, create the RC without specifying replicas, create the service, then scale up the RC, which may work better in an example using progressive disclosure and may have benefits in real scenarios also, such as ensuring one replica works before creating lots of them)
+1. Don't use `hostPort` unless absolutely necessary (e.g., for a node daemon) as it will prevent certain scheduling configurations due to port conflicts. Use the apiserver proxying or port forwarding for debug/admin access, or a service for external service access. If you need to expose a pod's port on the host machine, consider using a [NodePort](services.md#type--loadbalancer) service before resorting to `hostPort`.
+1. Don't use `hostNetwork` for the same reasons as `hostPort`.
+1. Don't specify default values unnecessarily to simplify and minimize configs. Now that we default (e.g. the selector and labels in ReplicationController), configs should feel slimmer.
+1. Service should use a subset of the labels used by the replication controllers whose pods it targets. This will allow you to update replication controllers without updating the services that target them and will make rolling-update more effective.
+1. Use kubectl bulk operations (via files and/or labels) for get and delete. See [label selectors](labels.md#label-selectors).
+1. Use kubectl run and expose to quickly create and expose single container replication controllers.
+1. Use headless services for easy service discovery when you don't need kube-proxy load balancing. See [headless services](services.md#headless-services).
+1. Use kubectl delete rather than stop. Delete has a super set of the functionality of stop and stop is deprecated.
+1. If there is a viable alternative to naked pods, use the alternative. Controllers are almost always preferable to creating pods (except for some RestartNever scenarios). A minimal Job is coming. See [#1624](https://github.com/GoogleCloudPlatform/kubernetes/issues/1743). Naked pods will not be rescheduled in the event of node failure.
+1. Put a version number or hash as a suffix to the name and in a label on a ReplicationController, to facilitate rolling update, as we do for [--image](https://github.com/GoogleCloudPlatform/kubernetes/blob/072137764665182394975e1e1459ade58ee25719/pkg/kubectl/rolling_updater.go#L112). This is necessary because rolling-update actually creates a new controller as opposed to modifying the existing controller. This does not play well with version agnostic controller names. Eventually you'll be able to use Deployment ([#1743](https://github.com/GoogleCloudPlatform/kubernetes/issues/1743)) for RestartAlways services rather than RC directly, when we have that.
```

Comment:
Reference the kubectl rolling-update doc rather than the code.

Don't mention Deployment yet, since we don't have it yet.